src/librustc_typeck/check/dropck.rs

```diff
@@ -95,16 +94,20 @@ fn ensure_drop_params_and_item_params_correspond<'tcx>(
             }
             Err(_) => {
                 let item_span = tcx.def_span(self_type_did);
+                let def_kind = tcx.def_kind(self_type_did).unwrap();
```

Comment:
Could you use `expect` or add a comment here, because it's not obvious why the `unwrap` is supposed to never fail.
Same for the `unwrap` below.

Reply:
Fixed by adding a fallback.